packages/graphql_server/src/api/twitterScrapingService.ts

```diff
@@ -0,0 +1,114 @@
+import axios, { AxiosRequestConfig } from 'axios'
+import {
+  TwitterPost,
+  TwitterUser,
+  TwitterProfileResponse
+} from '../../types/twitterScrapingService'
+
+const username = 'typescript2023'
+const apiKey = process.env.SCRAPING_BOT_API_KEY || ''
+const apiEndPoint = 'http://api.scraping-bot.io/scrape/data-scraper'
+const auth = 'Basic ' + Buffer.from(username + ':' + apiKey).toString('base64')
+
+const SORT_BY_RETWEETS = 'retweets'
+const SORT_BY_LIKES = 'likes'
+
+function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+/**
+ * Fetches the Twitter user by his handle
+ * @param twitterHandle
+ * @returns the TwitterUser with id, name and count of followers
+ */
+async function getTwitterUserByHandle(twitterHandle: string): Promise<TwitterUser> {
+  try {
+    // builds the axios request configuration
+    const requestConfig: AxiosRequestConfig = {
+      headers: {
+        Accept: 'application/json',
+        Authorization: auth
+      }
+    }
+    // specifies the used scraping mechanism, which is in this case twitterProfile
+    const requestData = {
+      scraper: 'twitterProfile',
+      url: 'https://twitter.com/' + twitterHandle
+    }
+
+    const response = await axios.post<{ responseId: string }>(
+      apiEndPoint,
+      requestData,
+      requestConfig
+    )
+
+    let data: TwitterProfileResponse | null = null
+
+    // sends the request until finalData is received
+    do {
+      await sleep(5000)
```

Comment:
Why do you wait for 5s here? If there's a good reason please comment it

Reply:
according the doc of the API you have to wait at least 5s here, otherwise we get blocked